### PR TITLE
Fix compiler warnings and update to Rizin 

### PR DESCRIPTION
**Pull request by officialcjunior**
## Description
Fixed all the existing warnings except this one:

```bash
python/core.c: In function ‘Rizin_plugin_core’:
python/core.c:67:19: error: ‘RzCorePlugin’ {aka ‘struct rz_core_plugin_t’} has no member named ‘call’
   67 |                 ap->call = py_core_call;
      |                   ^~
make[1]: *** [build.mk:16: lang_python.so] Error 1
make[1]: Leaving directory '/home/cjunior/Downloads/rz/rz-lang/python'
make: *** [Makefile:6: all] Error 2
```

This is because core plugins now use `RzCorePluginInit` and all. I'm not sure how to add this plugin
which doesn't have a (come under a) parent command tree and that's what I'm working on now.

### CODE REVIEW

**Review by wargio — APPROVED** [review]

**Comment by ret2libc** [comment]
We should probably convert to meson as well.

Apart from that, the problem with the `call` method seems quite important... I need to see better how the whole RzLang works though. Right now I don't have a solution to propose.

**Comment by ret2libc** [comment]
I think fixing `Rizin_plugin_core` is not straightforward at the moment. It should be made rz-shell-aware and probably it requires some changes here and there. I suggest to delete the core file for now and prevent to load RzCorePlugins from within RzLang. We can re-design that from scratch IMHO. cc @thestr4ng3r @XVilka @thestr4ng3r @wargio @kazarmy

**Comment by wargio** [comment]
Sounds like a good idea.

**Comment by XVilka** [comment]
ＬＧＴＭ　ｆｏｒ　ｒｅｍｏｖｉｎｇ　ｔｈｅ　ＲｚＣｏｒｅ　ｐｌｕｇｉｎ　ａｎｄ　ｃｏｎｖｅｒｔｉｎｇ　ｔｏ　ｔｈｅ　ｍｅｓｏｎ．

**Comment by officialcjunior** [comment]
```bash
$ make
for LANG in python; do make -C ${LANG} ; done
make[1]: Entering directory '/home/cjunior/Downloads/rz/rz-lang/python'
cc python.c python/*.c -I/usr/local/include/librz -I/usr/local/include/librz/sdb  -DPREFIX=\"/usr/local\" -I/usr/include/python3.9 -I/usr/include/python3.9 -DPYVER=3 -lpython3.9 -lcrypt -lpthread -ldl -lutil -lm -lm -L/usr/lib -lcrypt -lpthread -ldl -lutil -lm -lm -L/usr/local/lib  -lrz_core -lrz_io -lrz_util -shared -lrz_asm \
-I/usr/local/include/librz -I/usr/local/include/librz/sdb -L/usr/local/lib -lrz_core -lrz_lang -lrz_debug -lrz_hash -lrz_analysis -lrz_search -lrz_diff -lrz_type -lrz_egg -lrz_crypto -lrz_config -lrz_asm -lrz_parse -lrz_reg -lrz_flag -lrz_bin -lrz_magic -lrz_io -lrz_syscall -lrz_socket -lrz_bp -lrz_cons -lrz_util
\
 -L/usr/local/lib  -lrz_core -lrz_io -lrz_util -shared -lrz_asm -fPIC -o lang_python.so
make[1]: Leaving directory '/home/cjunior/Downloads/rz/rz-lang/python'
$
```
Compiles fine, now.

**Comment by XVilka** [comment]
Hmm:
```
ninja: Entering directory `build'
[0/1] Installing files.
Traceback (most recent call last):
  File "/home/runner/.local/bin/meson", line 5, in <module>
    from mesonbuild.mesonmain import main
ModuleNotFoundError: No module named 'mesonbuild'
FAILED: meson-install 
/home/runner/.local/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

**Comment by officialcjunior** [comment]
> Hmm:
> 
> ```
> ninja: Entering directory `build'
> [0/1] Installing files.
> Traceback (most recent call last):
>   File "/home/runner/.local/bin/meson", line 5, in <module>
>     from mesonbuild.mesonmain import main
> ModuleNotFoundError: No module named 'mesonbuild'
> FAILED: meson-install 
> /home/runner/.local/bin/meson install --no-rebuild
> ninja: build stopped: subcommand failed.
> ```

Wait, where is this from?

File: python/python/core.c [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-}
-
-void Rizin_plugin_core_free(RzCorePlugin *ap) {
-	free ((char *)ap->name);
-	free ((char *)ap->license);
-	free ((char *)ap->desc);
-	free (ap);
-}
-
-PyObject *Rizin_plugin_core(Rizin* self, PyObject *args) {
```
Comment (ret2libc) [inline python/python/core.c:54]: there's a reference to this function in python.c remove that as well.

Comment (officialcjunior) [reply python/python/core.c:54]: Removed that.

**Comment by XVilka** [comment]
@officialcjunior from CI: https://github.com/rizinorg/rz-lang/pull/1/checks?check_run_id=3027475379#step:5:2083

File: python/python/analysis.c [hunk 2]
```diff
@@ -125,7 +125,7 @@ void py_export_analysis_enum(PyObject *tp_dict) {
 		val->delta = getI (dict, "delta"); \
 		val->imm = getI (dict, "imm"); \
 		val->mul = getI (dict, "mul"); \
-		val->seg = getI (dict, "seg"); \
+		/* val->seg = getI (dict, "seg"); */ \
```
Comment (XVilka) [inline python/python/analysis.c]: Just remove this.

File: python/python/analysis.c [hunk 3]
```diff
@@ -169,7 +169,7 @@ static int py_analysis(RzAnalysis *a, RzAnalysisOp *op, ut64 addr, const ut8 *bu
 		};
 		PyObject *memview = PyMemoryView_FromBuffer (&pybuf);
 		PyObject *arglist = Py_BuildValue ("(NK)", memview, addr);
-		PyObject *result = PyEval_CallObject (py_analysis_cb, arglist);
+		PyObject *result = PyObject_CallObject (py_analysis_cb, arglist);
```
Comment (XVilka) [inline python/python/analysis.c]: I suggest to put `.clang-format` here - the same as in the main Rizin repo and reformat the code as well.

**Review by XVilka — APPROVED** [review]

File: .github/workflows/c-cpp.yml [hunk 4]
```diff
@@ -13,14 +13,13 @@ jobs:
     - name: Install python
       run: sudo apt-get --assume-yes install python3-wheel python3-setuptools python3-dev ninja-build
     - name: Install meson and ninja
-      run: pip3 install --user meson PyYAML
+      run: sudo pip3 install meson PyYAML ninja
     - uses: actions/checkout@v2
     - name: rizin
       run: |
-          git clone --recursive --depth=1 https://github.com/rizinorg/rizin
+          git clone --depth=1 https://github.com/rizinorg/rizin
```
Comment (wargio) [inline .github/workflows/c-cpp.yml:20]: do we want to have `stable` branch here? @XVilka  @ret2libc

Comment (ret2libc) [reply .github/workflows/c-cpp.yml:20]: For consistency we may use the same approach everywhere

Comment (wargio) [reply .github/workflows/c-cpp.yml:20]: which is?

Comment (XVilka) [reply .github/workflows/c-cpp.yml:20]: @wargio not for now. We could do the switch after 0.3.0 is out, and RzLang properly fixed.

**Pull request merged**